nipoppy/workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -489,23 +491,25 @@ def run(participant_id, global_configs, session_id, output_dir, use_bids_filter,
             ## check that vectors are unique
             tvec = bvec[:,tndir]
             tdir = np.unique(tvec, axis=0)
-
-            ## compute and print the maximum shell - can get odd numbers?
-            tlmax = int(np.floor((-3 + np.sqrt(1 + 8 * tdir.shape[1])) / 2.0))
+            mldir.append(tdir.shape[1])
+            
+            ## compute and print the maximum shell
+            tlmax = int(np.floor((-3 + np.sqrt(1 + 8 * tdir.shape[1]) / 2.0)))
             mlmax.append(tlmax)
             logger.info(f" -- Shell {int(shell)} has {tdir.shape[1]} directions capable of a max lmax: {tlmax}")
 
-        ## the minimum lmax across shells is used
-        plmax = min(mlmax)
-        logger.info(f"The maximum lmax across shells is: {plmax}")
+        ## the max lmax within any 1 shell is used
+        plmax = max(mlmax)
+        logger.info(f"The maximum lmax for any one shell is: {plmax}")
         
     ## if lmax too large, reset with warning
     if sh_order <= plmax:
         logger.info(f"Running model with requested lmax: {sh_order}")
     else:
```

Comment:
This block only has logging?

Reply:
Yeah. This is where I would have overrode a bad lmax. But the logic wasn't working with the new data, so I decided it was easier to just print some useful debugging in case it is an issue later on.